susemanager-utils/testing/docker/scripts/schema_idempotency_test_pgsql.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/python
+import argparse
+import difflib
+import distutils.version as DV
+import json
+import os
+import re
+import shutil
+import subprocess
+import sys
+
+
+def file_exists(path):
+    try:
+        open(path, 'r')
+        return(True)
+    except FileNotFoundError:
+        return(False)
+
+
+def directory_exists(path):
+    if os.path.isdir(path):
+        return(True)
+    return(False)
+
+
+def get_all_files_rpm_pr(prFile):
+    with open(prFile, 'r') as f:
+        changedFiles = []
+        files = json.loads(read(f))['files']
+        for file in filesDict:
+            if file['status'] == 'modified' or file['status'] == 'added':
+                changedFiles.append(file['filename'])
+    return(changedFiles)
+
+
+def get_all_files_since(version, schemaPath):
+    files = []
+    for element in os.listdir(schemaPath):
+        elementVer = re.search(
+            '^susemanager-schema-(\d+\.\d+[\d|.]*)-to-susemanager-schema-\d+\.\d+[\d|.]*$', element)
+        if elementVer:
```

Comment:
i would use here GuardeClause.

`if ! elementVer: continue`

So you gain simplity of code ( nested if add also complexity) and is difficult to read

Reply:
Fixed and testing. One of this days I will be able to remember about it :-D